libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -163,6 +168,44 @@ impl<'a> Validator<'a> {
         errors.ok()
     }
 
+    fn validate_names_for_indexes(
+        &self,
+        ast_schema: &ast::SchemaAst,
+        schema: &dml::Datamodel,
+    ) -> Result<(), ErrorCollection> {
+        let mut errors = ErrorCollection::new();
+        let mut index_names = HashSet::new();
+
+        let multiple_indexes_with_same_name_are_supported = match self.source {
+            Some(source) => source.combined_connector.supports_multiple_indexes_with_same_name(),
+            None => false,
+        };
+
+        for model in schema.models() {
+            if let Some(ast_model) = ast_schema.find_model(&model.name) {
+                for index in model.indices.iter() {
+                    if let Some(index_name) = &index.name {
+                        if index_names.contains(index_name) && !multiple_indexes_with_same_name_are_supported {
+                            let ast_index = ast_model
+                                .directives
+                                .iter()
+                                .find(|directive| directive.name.name == "index")
+                                .unwrap();
```

Comment:
Do we want to panic here with no error? Could there be a case where we do not have a directive named `index`?

Reply:
No need to change if we cannot panic here. Just be sure about it. :)

Reply:
Normally there isn't a case where we do not have a directive named `index` at this point. 
I can instead put it inside a `if let  ..`, but most other functions inside this file do the same thing, that's why I used those as a reference.

Reply:
Do you think a missing directive here would be an error and we should crash? If so, please `unwrap`, if not and we can continue execution, `if let` is the right choice.

Reply:
Got it, I'll use `unwrap` then 👍